File: cpp/src/stats.cpp
#include <clankers/stats.hpp>

namespace clankers {

InferenceStats InferenceStats::from_c(const ClankersInferenceStats& raw) {
    InferenceStats stats;
    stats.latency_us = raw.latency_us;
    stats.copies = raw.copies;
    stats.bytes_copied = raw.bytes_copied;
    stats.allocations = raw.allocations;
    stats.bytes_allocated = raw.bytes_allocated;
    stats.backend_latency_us = raw.backend_latency_us;
    stats.backend_copies = raw.backend_copies;
    stats.backend_bytes_copied = raw.backend_bytes_copied;
    return stats;
}

}  // namespace clankers

File: crates/clankers-ffi/src/stats.rs
//! [`ClankersInferenceStats`] conversions.

use clankers_ml::inference::InferenceStats;

use crate::ClankersInferenceStats;

pub fn stats_to_c(stats: &InferenceStats) -> ClankersInferenceStats {
    ClankersInferenceStats {
        latency_us: stats.latency.as_micros() as u64,
        copies: stats.clankers_copies,
        bytes_copied: stats.clankers_bytes_copied,
        allocations: stats.allocations,
        bytes_allocated: stats.bytes_allocated,
        backend_latency_us: 0,
        backend_copies: stats.backend.backend_copies,
        backend_bytes_copied: stats.backend.backend_bytes_copied,
    }
}
